Automattic/pocket-casts-android
# Simplify Analytics configuration

## Description

I'm doing this work to make my work with clip analytics easier.

We had a strange configuration of `object AnalyticsTracker` and `class AnalyticsTrackerWrapper`. The first one was used to register all trackers and the latter one was used for DI but it was a bit clunky to use because it called `object AnalyticsTracker` inside.

This PR removes `object AnalyticsTracker` and uses proper DI for `class AnalyticsTracker`. Previously DI was impossible because there was a cyclic dependency, which I had to break in this PR by introducing `AccountManagerStatusInfo`. The PR also decouples it from `Settings` making testing a bit easier without a need for mocks.

You can ignore 6605bbae3fc67a613482d80d048ba8c0af216d7c as it is just a renaming PR. The meat of the PR is in f399dc90e5531fbaf6b3d2e33c4fd5807a81bf89.

## Testing Instructions

1. Open the app as the unsigned user and verify that tracking works.
2. Analytic events should have `"user_is_logged_in":false` property.
3. Sign in.
4. Analytic events should have `"user_is_logged_in":true` property. Signing out at this point will fail to reflect the property change (#2409).
5. You can verify in the network inspector that user ID (`_ui`) is still correctly tracked. However user type will be incorrect due to this bug - https://github.com/Automattic/Automattic-Tracks-Android/pull/223.
![Screenshot 2024-06-28 at 10 43 33](https://github.com/Automattic/pocket-casts-android/assets/30936061/a0b657d7-2e92-44b2-9185-e0cd728c0325)
7. Go to privacy settings and disable `Analytics`.
8. Notice that events are no longer tracked.

## Checklist
 
#### I have tested any UI changes...
